src/components/admin/ContentRenderer.tsx

```diff
@@ -86,24 +85,52 @@ export const getMetadata = async (contentId: number) => {
     };
   }
 
-  const mainUrls = {
-    1080: metadata[`video_1080p_mp4_${userId}`],
-    720: metadata[`video_720p_mp4_${userId}`],
-    360: metadata[`video_360p_mp4_${userId}`],
-    subtitles: metadata['subtitles'],
-    slides: metadata['slides'],
-    segments: metadata['segments'],
-  };
+  // Check the highest quality video URL (1080p) first
+  const highestQualityUrl = metadata[`video_1080p_mp4_${userId}`];
+  const isHighestQualityUrlAccessible = await isUrlAccessible(highestQualityUrl);
 
-  const isMainUrlAccessible = await Promise.all(
-    Object.values(mainUrls).map(isUrlAccessible)
-  );
+  if (isHighestQualityUrlAccessible) {
+    // If the highest quality URL is accessible, return all main URLs
+    return {
+      1080: highestQualityUrl,
+      720: metadata[`video_720p_mp4_${userId}`],
+      360: metadata[`video_360p_mp4_${userId}`],
+      subtitles: metadata['subtitles'],
+      slides: metadata['slides'],
+      segments: metadata['segments'],
+    };
+  }
+
+  // If the highest quality URL is not accessible, check the 720p URL
+  const mediumQualityUrl = metadata[`video_720p_mp4_${userId}`];
```

Comment:
can we improve the code?

```
const mainUrls = {
    1080: metadata[`video_1080p_mp4_${userId}`],
    720: metadata[`video_720p_mp4_${userId}`],
    360: metadata[`video_360p_mp4_${userId}`],
    subtitles: metadata['subtitles'],
    slides: metadata['slides'],
    segments: metadata['segments'],
    thumbnails: metadata['thumbnail_mosiac_url'],
  };

  const bunnyUrls = {
    1080: bunnyUrl(metadata[`video_1080p_mp4_${userId}`]),
    720: bunnyUrl(metadata[`video_720p_mp4_${userId}`]),
    360: bunnyUrl(metadata[`video_360p_mp4_${userId}`]),
    subtitles: metadata['subtitles'],
    slides: metadata['slides'],
    segments: metadata['segments'],
    thumbnails: metadata['thumbnail_mosiac_url'],
  };

  const isHighestQualityUrlAccessible = await isUrlAccessible(mainUrls['1080']);

  if (isHighestQualityUrlAccessible) {
    return mainUrls;
  }

  const otherQualities = ['720', '360'];
  for (const quality of otherQualities) {
    const urlKey = `${quality}`;
    const isAccessible = await isUrlAccessible(mainUrls[urlKey]);
    if (isAccessible) {
      return mainUrls;
    }
  }

  // If none of the main URLs are accessible, return Bunny URLs
  return bunnyUrls;
```


something like this?

Reply:
> can we improve the code?
> 
> ```
> const mainUrls = {
>     1080: metadata[`video_1080p_mp4_${userId}`],
>     720: metadata[`video_720p_mp4_${userId}`],
>     360: metadata[`video_360p_mp4_${userId}`],
>     subtitles: metadata['subtitles'],
>     slides: metadata['slides'],
>     segments: metadata['segments'],
>     thumbnails: metadata['thumbnail_mosiac_url'],
>   };
> 
>   const bunnyUrls = {
>     1080: bunnyUrl(metadata[`video_1080p_mp4_${userId}`]),
>     720: bunnyUrl(metadata[`video_720p_mp4_${userId}`]),
>     360: bunnyUrl(metadata[`video_360p_mp4_${userId}`]),
>     subtitles: metadata['subtitles'],
>     slides: metadata['slides'],
>     segments: metadata['segments'],
>     thumbnails: metadata['thumbnail_mosiac_url'],
>   };
> 
>   const isHighestQualityUrlAccessible = await isUrlAccessible(mainUrls['1080']);
> 
>   if (isHighestQualityUrlAccessible) {
>     return mainUrls;
>   }
> 
>   const otherQualities = ['720', '360'];
>   for (const quality of otherQualities) {
>     const urlKey = `${quality}`;
>     const isAccessible = await isUrlAccessible(mainUrls[urlKey]);
>     if (isAccessible) {
>       return mainUrls;
>     }
>   }
> 
>   // If none of the main URLs are accessible, return Bunny URLs
>   return bunnyUrls;
> ```
> 
> something like this?

sure , thats cleaner lol